packages/flutter_tools/test/devfs_test.dart

```diff
@@ -39,18 +40,21 @@ void main() {
       await devFS.update();
       expect(devFSOperations.contains('writeFile test bar/foo.txt'), isTrue);
     });
-    testUsingContext('modify existing file on local file system', () async {
-      File file = new File(path.join(basePath, filePath));
-      file.writeAsBytesSync(<int>[1, 2, 3, 4, 5, 6]);
-      await devFS.update();
-      expect(devFSOperations.contains('writeFile test bar/foo.txt'), isTrue);
-    });
     testUsingContext('add new file to local file system', () async {
       File file = new File(path.join(basePath, filePath2));
       await file.parent.create(recursive: true);
       file.writeAsBytesSync(<int>[1, 2, 3, 4, 5, 6, 7]);
       await devFS.update();
       expect(devFSOperations.contains('writeFile test foo/bar.txt'), isTrue);
+      // Need to delay between when we first write to a file and when
+      // we modify it.
+      await new Future<Null>.delayed(new Duration(seconds: 1));
```

Comment:
The underyling problem is that file modification timestamps on MacOS are not high resolution enough. The flow in the test is:

1) Write a new file.
2) Modify the file.
3) Verify that we detect the modification.

Putting a sleep between (1) and (2) makes the mtimes different.